### ISSUE TITLE
Pay upon Invoice all orders status "on hold"

### ISSUE THREAD

**Issue by originalcandleguy** [woocommerce/woocommerce-paypal-payments#2860]
### Describe the Bug

When a customer orders with the payment method "Pay upon Invoice" all orders have the order status "on hold". 

### To Reproduce

1. Place an order
2. Pay with "Pay upon invoice"
3. Login to the backend
4. Check the oroder
5. It will be on hold

### Expected Behavior

All orders will have the status "processing" with the payment method "pay upon invoice"

### Actual Behavior

All orders have the status "on hold" with the payment method "pay upon invoice"

### Environment

 - Wordpress Version 6.7.1
 - WooCommerce Version 9.4.2
 - Plugin Version 2.9.4
 - Chrome

### Additional Details

<details><summary>System status</summary>

  ```
<details><summary>System Status Report</summary>

```
### WordPress Environment ###

WordPress address (URL): [Redacted]
Site address (URL): [Redacted]
WC Version: 9.4.2
Legacy REST API Package Version: The Legacy REST API plugin is not installed on this site.
Action Scheduler Version: ✔ 3.8.2
Log Directory Writable: ✔
WP Version: 6.7.1
WP Multisite: –
WP Memory Limit: 512 MB
WP Debug Mode: –
WP Cron: –
Language: de_DE
External object cache: –

### Server Environment ###

Server Info: Apache/2.4.62 (Debian)
PHP Version: 8.1.30
PHP Post Max Size: 50 MB
PHP Time Limit: 600
PHP Max Input Vars: 2500
cURL Version: 7.74.0
OpenSSL/1.1.1w

SUHOSIN Installed: –
MySQL Version: 10.5.27-MariaDB-deb11-log
Max Upload Size: 50 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

[REDACTED]

### Post Type Counts ###

attachment: 2372
blocks: 28
custom_css: 1
customize_changeset: 11
document_template: 6
nav_menu_item: 75
page: 49
post: 3
product: 629
product_variation: 218
revision: 807
shop_coupon: 105
shop_order: 41917
shop_order_refund: 17
ux_template: 2
wp_global_styles: 1
wp_navigation: 1
wpcf7_contact_form: 2

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (25) ###

AIOSEO – Weiterleitungen: by All in One SEO Team – 1.4.8
All in One SEO Pro: by All in One SEO Team – 4.7.6
Breeze: by Cloudways – 2.1.19
Contact Form 7: by Takayuki Miyoshi – 6.0.1
Easy Product Bundles for WooCommerce: by Product Bundles Team – 5.5.0
WooGallery: by WooGallery Team
ShapedPlugin LLC – 2.1.5

Loco Translate: by Tim Whitlock – 2.6.14
Mindestkaufbetrag für Woo-Warenkorb: by CT-Gespräche – 2.3.19
Object Cache Pro: by Rhubarb Group – 1.21.3
Omnisend for WooCommerce: by Omnisend – 1.15.22
One Stop Shop für WooCommerce: by vendidero – 1.6.2
Conversios.io - Product Feed Manager for WooCommerce: by Conversios – 4.0.0
Product Video Gallery for Woocommerce: by NikHiL Gadhiya – 1.4.2.8
Two-Factor: by Mitwirkende – 0.9.1
Import Export WordPress Users and WooCommerce Customers: by WebToffee – 2.5.9
J Cart Upsell and Cross-sell for WooCommerce: by Giacomo Zoffoli – 3.8.2
Weglot Translate: by Weglot Translate team – 4.2.9
Payment Plugins for Stripe WooCommerce: by Payment Plugins
support@paymentplugins.com – 3.3.78

Germanized für WooCommerce Pro: by vendidero – 4.1.3
Germanized für WooCommerce: by vendidero – 3.18.5
WooCommerce PayPal Payments: by WooCommerce – 2.9.4
WooCommerce: by Automattic – 9.4.2
WP Mail SMTP: by WP Mail SMTP – 4.2.0
WPS Hide Login: by WPServeur
NicolasKulka
wpformation – 1.9.17.1

YITH WooCommerce Wunschliste: by YITH – 4.0.1

### Dropin Plugins () ###

advanced-cache.php: advanced-cache.php

### Settings ###

Legacy API Enabled: –
Force SSL: –
Currency: EUR (€)
Currency Position: right_space
Thousand Separator: .
Decimal Separator: ,
Number of Decimals: 2
Taxonomies: Product Types: easy_product_bundle (easy_product_bundle)
external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –
Enforce Approved Product Download Directories: ✔
HPOS feature enabled: –
Order datastore: WC_Order_Data_Store_CPT
HPOS data sync enabled: –

### Logging ###

Enabled: ✔
Handler: Automattic\WooCommerce\Internal\Admin\Logging\LogHandlerFileV2
Retention period: 30 days
Level threshold: –
Log directory size: 11 MB

### WC Pages ###

Shop base: #6 - /shop
Cart: #7 - /cart -  Contains the [woocommerce_cart] shortcode
Checkout: #8 - /checkout -  Contains the [woocommerce_checkout] shortcode
My account: #9 - /my-account
Terms and conditions: #169608 - /agb

### Theme ###

Name: Flatsome Child
Version: 3.0
Author URL: 
Child Theme: ✔
Parent Theme Name: Flatsome
Parent Theme Version: 3.19.7
Parent Theme Author URL: https://uxthemes.com
WooCommerce Support: ✔

### Templates ###

Overrides: flatsome/woocommerce/archive-product.php
flatsome/woocommerce/cart/cart-empty.php
flatsome/woocommerce/cart/cart-shipping.php
flatsome/woocommerce/cart/cart.php
flatsome/woocommerce/cart/mini-cart.php
flatsome/woocommerce/checkout/form-checkout.php
flatsome/woocommerce/checkout/form-coupon.php
flatsome/woocommerce/checkout/order-received.php
flatsome/woocommerce/checkout/terms.php
flatsome/woocommerce/checkout/thankyou.php
flatsome/woocommerce/content-product-cat.php
flatsome/woocommerce/content-product.php
flatsome/woocommerce/content-single-product.php
flatsome/woocommerce/content-widget-product.php
flatsome-child/woocommerce/emails/customer-processing-order.php
flatsome/woocommerce/global/breadcrumb.php
flatsome/woocommerce/global/quantity-input.php
flatsome/woocommerce/global/wrapper-end.php
flatsome/woocommerce/global/wrapper-start.php
flatsome/woocommerce/loop/header.php
flatsome/woocommerce/loop/loop-end.php
flatsome/woocommerce/loop/loop-start.php
flatsome/woocommerce/loop/pagination.php
flatsome/woocommerce/loop/rating.php
flatsome/woocommerce/loop/result-count.php
flatsome/woocommerce/loop/sale-flash.php
flatsome/woocommerce/myaccount/form-login.php
flatsome/woocommerce/myaccount/navigation.php
flatsome/woocommerce/notices/error.php
flatsome/woocommerce/notices/notice.php
flatsome/woocommerce/notices/success.php
flatsome/woocommerce/product-searchform.php
flatsome/woocommerce/single-product/photoswipe.php
flatsome/woocommerce/single-product/price.php
flatsome/woocommerce/single-product/product-image.php
flatsome/woocommerce/single-product/product-thumbnails.php
flatsome/woocommerce/single-product/rating.php
flatsome/woocommerce/single-product/related.php
flatsome/woocommerce/single-product/review.php
flatsome/woocommerce/single-product/sale-flash.php
flatsome/woocommerce/single-product/share.php
flatsome/woocommerce/single-product/short-description.php
flatsome/woocommerce/single-product/tabs/tabs.php
flatsome/woocommerce/single-product/title.php
flatsome/woocommerce/single-product/up-sells.php
flatsome/woocommerce/single-product-reviews.php
flatsome/woocommerce/single-product.php

### WooCommerce PayPal Payments ###

Onboarded: ✔
Shop country code: DE
WooCommerce currency supported: ✔
Advanced Card Processing available in country: ✔
Pay Later messaging available in country: ✔
Webhook status: ✔
PayPal Vault enabled: –
ACDC Vault enabled: –
Logging enabled: –
Reference Transactions: –
Used PayPal Checkout plugin: –
Subscriptions Mode: Disabled
PayPal Shipping Callback: ✔
Apple Pay: –
Google Pay: –
Fastlane: –

### Admin ###

Enabled Features: activity-panels
analytics
product-block-editor
coupons
core-profiler
customize-store
customer-effort-score-tracks
import-products-task
experimental-fashion-sample-products
shipping-smart-defaults
shipping-setting-tour
homescreen
marketing
mobile-app-banner
navigation
onboarding
onboarding-tasks
pattern-toolkit-full-composability
product-custom-fields
remote-inbox-notifications
remote-free-extensions
payment-gateway-suggestions
printful
shipping-label-banner
subscriptions
store-alerts
transient-notices
woo-mobile-welcome
wc-pay-promotion
wc-pay-welcome-page
launch-your-store

Disabled Features: product-data-views
experimental-blocks
minified-js
product-pre-publish-modal
settings
async-product-editor-category-field
product-editor-template-system
blueprint
reactify-classic-payments-settings
use-wp-horizon

Daily Cron: ✔ Next scheduled: 2024-11-29 12:15:20 +00:00
Options: ✔
Notes: 103
Onboarding: skipped

### Action Scheduler ###

Complete: 6,306
Oldest: 2024-10-29 08:46:30 +0000
Newest: 2024-11-29 05:58:36 +0000

Failed: 21
Oldest: 2023-11-14 12:21:35 +0000
Newest: 2024-11-20 13:20:07 +0000

Pending: 19
Oldest: 2024-11-29 06:09:03 +0000
Newest: 2024-12-03 06:31:29 +0000

### Status report information ###

Generated at: 2024-11-29 06:07:53 +00:00
```
</details>
  ```

</details>

**Comment by SydeKrystian** [comment]
Hello @originalcandleguy 

It seems that your GitHub issue is primarily a support request. While we value your input and feedback, GitHub is predominantly used for our development activities. For a prompt and detailed response, we kindly ask that you reach out directly to our [support team](https://woocommerce.com/document/woocommerce-paypal-payments/#get-help).  

When you contact our support team, please include a copy of your [WooComnmerce system report](https://woocommerce.com/document/understanding-the-woocommerce-system-status-report/). This will help us get to the bottom of your issue more quickly. 

We'll be closing this GitHub issue for now. However, once you reach out to our support team, rest assured they'll be ready to assist you with your concerns. Thank you for your understanding!

Kind regards,
Krystian
